src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +238,157 @@ public class PatientEverythingService : IPatientEverythingService
+
+            List<Patient.LinkComponent> links = ExtractLinksFromParentPatient(parentPatientResource);
+
+            if (links == null)
+            {
+                return null;
+            }
+
+            var seeAlsoLinkIdsHash = new HashSet<string>();
+
+            // Walk through the links, storing the "seealso" links to a hash set.
+            // This will allow us to avoid:
+            // 1. Links that aren't "seealso" links
+            // 2. Links that don't point to Patient resources
+            // 3. Duplicates
+            foreach (Patient.LinkComponent link in links)
+            {
+                ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                if (link.Type != Patient.LinkType.Seealso)
+                {
+                    continue;
+                }
+
+                // If the link points back to the parent patient
+                if (string.Equals(referenceSearchValue.ResourceId, parentPatientId, StringComparison.Ordinal))
+                {
+                    // Ignore it to avoid running patient $everything on the same patient again.
+                    continue;
+                }
+
+                // Links can be of type Patient or RelatedPerson. Only follow the links that point to patients.
+                // Note: we plan to include RelatedPerson resources in the result set in AB#85142.
+                if (string.Equals(referenceSearchValue.ResourceType, ResourceType.Patient.ToString(), StringComparison.Ordinal))
+                {
+                    if (!seeAlsoLinkIdsHash.Contains(referenceSearchValue.ResourceId))
+                    {
+                        seeAlsoLinkIdsHash.Add(referenceSearchValue.ResourceId);
+                    }
+                }
+            }
+
+            if (seeAlsoLinkIdsHash.Count == 0)
+            {
+                // The parent patient has links, but no "seealso" patient links.
+                return null;
+            }
+
+            // Sort the list to ensure that it is in the same order each time we access it.
+            var seeAlsoLinkIdsSorted = seeAlsoLinkIdsHash.ToList();
+            seeAlsoLinkIdsSorted.Sort();
+
+            // If we haven't processed any "seealso" links yet
+            if (token.CurrentSeeAlsoLinkId == null)
+            {
+                // Then the next "seealso" link we process will be the first in the list.
+                token.CurrentSeeAlsoLinkId = seeAlsoLinkIdsSorted.First();
+            }
+            else
+            {
+                // Otherwise, the next "seealso" link we process will be the one that follows the current one in the sorted list.
+                int indexOfCurrent = seeAlsoLinkIdsSorted.FindIndex(id => id == token.CurrentSeeAlsoLinkId);
+                int indexOfNext = indexOfCurrent + 1;
+
+                if (indexOfNext < seeAlsoLinkIdsSorted.Count - 1)
```

Comment:
My understanding is that the condition should be `indexOfNext < seeAlsoLinkIdsSorted.Count`. Otherwise we will lose the last "seealso" link.

Reply:
Thanks for catching this! Fixed in 6b9eba8 and added a test for it.